main.py

```diff
@@ -152,6 +182,9 @@ def pattern_match(patterns, source_list):
             task_names.add(matching)
     return list(task_names)
 
+def get_gpus_max_memory(max_memory):
+    max_memory = {i: max_memory for i in range(torch.cuda.device_count())}
```

Comment:
Added a warning 👍

Reply:
we could also fix it by computing `max_memory` after the accelerator is defined and using `accelerator.num_processes` instead of `torch.cuda.device_count()`

Reply:
Nice done!